docs/reference/rise-primitives.md

```diff
@@ -0,0 +1,37 @@
+---
+id: rise-primitives
+title: RISE Primitives
+sidebar_label: RISE Primitives
+slug: /reference/rise-primitives
+---
+
+## Type syntax cheatsheet
+
+Type | Syntax
+-----|-------
+Function from `A` to `B` | `(A -> B)`
+Dependent function from `a : A` to `B a` | `(x : A -> B)`
+Dependent function application | `f(n)`
+Vector of `A` of size `n` | `<n>A`
+Array of `A` of size `n` | `n.A`
+Dependent array of `F : Nat -> A` of size `n` | `n..F`
+Index of size `n : Nat` | `idx[n]`
+Pair of `A` and `B` | `(A, B)`
+Dependent pair of `x : A` and `B a` | `(x : A ** B)`
+
+## Primitives
+
+```scala mdoc:passthrough
+import rise.core.{primitives, makeClosed}
+println("Primitive | Type")
+println("----------|-----")
+primitives.members.foreach { p =>
+  var ts = p.typeScheme
+  // Try and close the implicit identifiers in the types
+  try { ts = makeClosed(ts)._1 }
+    catch { case _ : Throwable => () }
+  // Prevent types from being interpreted as column separators
+  val tsrepr = ts.toString.replace("|", "")
```

Comment:
This is a bit delicate. With this line here you change the meaning of some types.
Take `depZip` as an example. The type should be:
```
depZip: (n: nat -> (n.._n2d1 -> (n.._n2d2 -> n..(m: nat |-> (_n2d1(m), _n2d2(m))))))
                                                 ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
                                                 This indicates that this is a type-level
                                                 anonymous function, not a dependent function type
```
Replacing the `|->` to a `->` changes the kind of the underlined expression.
Maybe we can find another notation to make clear that this arrow is different?

Reply:
Is it possible to escape this symbol, using `&#124;` for example?

Reply:
I tried a bunch of different things:
- Using `|` inside the code tags is still interpreted as "next table cell"
- Escaping with `\|` works outside of the code tags but not inside, as the backslash is automatically escaped, i.e. we get `\\|` that is rendered as `\|`.
- Escaping with `&#124;` works outside of the code tags but not inside, the ampersand is automatically escaped, i.e. we get `\&#124;` which is rendered as `&#124;`.
- Replacing `|->` with the unicode symbol `⟼` works but it looks kind of small.
- Replacing `|` with `⊢` looks good and gives `⊢->`.